File: app/src/main/java/cn/intret/app/picgo/model/LoadMediaFileParam.java
package cn.intret.app.picgo.model;

public class LoadMediaFileParam {
    boolean mFromCacheFirst = true;
    boolean mLoadMediaInfo = false;
    SortWay mSortWay = SortWay.DATE;
    SortOrder mSortOrder = SortOrder.DESC;

    public boolean isFromCacheFirst() {
        return mFromCacheFirst;
    }

    public LoadMediaFileParam setFromCacheFirst(boolean fromCacheFirst) {
        mFromCacheFirst = fromCacheFirst;
        return this;
    }

    public boolean isLoadMediaInfo() {
        return mLoadMediaInfo;
    }

    public LoadMediaFileParam setLoadMediaInfo(boolean loadMediaInfo) {
        mLoadMediaInfo = loadMediaInfo;
        return this;
    }

    public SortWay getSortWay() {
        return mSortWay;
    }

    public LoadMediaFileParam setSortWay(SortWay sortWay) {
        mSortWay = sortWay;
        return this;
    }

    public SortOrder getSortOrder() {
        return mSortOrder;
    }

    public LoadMediaFileParam setSortOrder(SortOrder sortOrder) {
        mSortOrder = sortOrder;
        return this;
    }
}

File: app/src/main/java/cn/intret/app/picgo/model/image/data/LoadMediaFileParam.kt
package cn.intret.app.picgo.model.image.data

import cn.intret.app.picgo.model.user.data.SortOrder
import cn.intret.app.picgo.model.user.data.SortWay

class LoadMediaFileParam {
    internal var mFromCacheFirst = true
    internal var mLoadMediaInfo = false
    internal var mSortWay = SortWay.DATE
    internal var mSortOrder = SortOrder.DESC

    fun isFromCacheFirst(): Boolean {
        return mFromCacheFirst
    }

    fun setFromCacheFirst(fromCacheFirst: Boolean): LoadMediaFileParam {
        mFromCacheFirst = fromCacheFirst
        return this
    }

    fun isLoadMediaInfo(): Boolean {
        return mLoadMediaInfo
    }

    fun setLoadMediaInfo(loadMediaInfo: Boolean): LoadMediaFileParam {
        mLoadMediaInfo = loadMediaInfo
        return this
    }

    fun getSortWay(): SortWay {
        return mSortWay
    }

    fun setSortWay(sortWay: SortWay): LoadMediaFileParam {
        mSortWay = sortWay
        return this
    }

    fun getSortOrder(): SortOrder {
        return mSortOrder
    }

    fun setSortOrder(sortOrder: SortOrder): LoadMediaFileParam {
        mSortOrder = sortOrder
        return this
    }

    override fun toString(): String {
        return "LoadMediaFileParam{" +
                "mFromCacheFirst=" + mFromCacheFirst +
                ", mLoadMediaInfo=" + mLoadMediaInfo +
                ", sortWay=" + mSortWay +
                ", sortOrder=" + mSortOrder +
                '}'.toString()
    }
}
